packages/core/src/version.ts

```diff
@@ -4,13 +4,15 @@ import cbor from 'cbor';
 export interface Version {
   withMetadata: string;
   withoutMetadata: string;
+  linkedWithoutMetadata: string;
 }
 
-export function getVersion(bytecode: string): Version {
+export function getVersion(bytecode: string, linkedBytecode?: string): Version {
   if (bytecode !== '') {
     return {
       withMetadata: hashBytecode(bytecode),
       withoutMetadata: hashBytecodeWithoutMetadata(bytecode),
+      linkedWithoutMetadata: hashBytecodeWithoutMetadata(linkedBytecode ?? bytecode),
```

Comment:
I think it would be better to not default to `bytecode` and only set this if the `linkedBytecode` argument is present.

Reply:
The issue if we make `linkedWithoutMetadata` optional is that then we need to do additional validations when calling `fetchOrDeploy` as this is the parameter that is passed and it is clearly mandatory. 

One workaround would bet o do something like this:

```javascript
  return fetchOrDeployGeneric(implLens(version.linkedWithoutMetadata ?? version.withoutMetadata), provider, deploy);
```
What do you think?

Reply:
If we proceed this is how `Version` and `getVersion` look like:

```typescript
export interface Version {
  withMetadata: string;
  withoutMetadata: string;
  linkedWithoutMetadata?: string;
}

export function getVersion(bytecode: string, linkedBytecode?: string): Version {
  if (bytecode !== '') {
    const version: Version = {
      withMetadata: hashBytecode(bytecode),
      withoutMetadata: hashBytecodeWithoutMetadata(bytecode),
    };
    if (linkedBytecode) {
      version.linkedWithoutMetadata = hashBytecodeWithoutMetadata(linkedBytecode);
    }

    return version;
  } else {
    throw new Error('Abstract contract not allowed here');
  }
}
```

Reply:
Hm, that's true. Let's keep it then!

Reply:
Another option is to return it empty (but defined) unless the parameter is specified and in that case it uses the hash.